cinema/views.py

```diff
@@ -94,13 +97,34 @@ def get_queryset(self):
 
         return queryset.distinct()
 
+    @action(
+        methods=["POST"],
+        detail=TokenAuthentication,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid(raise_exception=True):
```

Comment:
you don't need `if` and `return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)` if you use `serializer.is_valid(raise_exception=True)` . serialzer returns it automatically